### PR TITLE
Compatibility with fbghc

### PR DESCRIPTION
**Pull request by pepeiborra**
Rather than forking ghcide, we use conditional compilation to build with https://github.com/facebook/fbghc

### CODE REVIEW

**Review by wz1000 — APPROVED** [review]

**Review by ndmitchell — APPROVED** [review]

File: src/Development/IDE/GHC/Compat.hs [hunk 1]
```diff
@@ -271,6 +273,13 @@ applyPluginsParsedResultAction :: HscEnv -> DynFlags -> ModSummary -> ApiAnns ->
...
   -- Apply parsedResultAction of plugins
   let applyPluginAction p opts = parsedResultAction p opts ms
-  fmap hpm_module $ 
-    runHsc env $ withPlugins dflags applyPluginAction 
+  fmap hpm_module $
+    runHsc env $ withPlugins dflags applyPluginAction
       (HsParsedModule parsed [] hpm_annotations)
+
+pattern ExposePackage :: String -> PackageArg -> ModRenaming -> PackageFlag
+#ifdef __FACEBOOK_HASKELL__
```
Comment (ndmitchell) [inline src/Development/IDE/GHC/Compat.hs:280]: Could we get a link to the issue, or the fbghc repo, so its clear what this means.

Comment (pepeiborra) [reply src/Development/IDE/GHC/Compat.hs]: good point, added

**Pull request merged**